cg/meta/workflow/taxprofiler.py

```diff
@@ -18,3 +34,85 @@ def __init__(
         pipeline: Pipeline = Pipeline.TAXPROFILER,
     ):
         super().__init__(config=config, pipeline=pipeline)
+        self.root_dir = config.taxprofiler.root
+
+    @property
+    def root(self) -> str:
+        return self.root_dir
+
+    @property
+    def fastq_handler(self):
+        return TaxprofilerFastqHandler
+
+    def get_case_config_path(self, case_id):
+        return NextflowAnalysisAPI.get_case_config_path(case_id=case_id, root_dir=self.root_dir)
+
+    @staticmethod
+    def build_samplesheet_content(
+        case_id: str,
+        fastq_r1: List[str],
+        fastq_r2: List[str],
+        instrument_platform: str,
+        fasta: Optional[str],
+    ) -> Dict[str, List[str]]:
+        """Build samplesheet headers and lists"""
+        try:
+            TaxprofilerSample(
+                sample=case_id,
+                fastq_r1=fastq_r1,
+                fastq_r2=fastq_r2,
+                instrument_platform=instrument_platform,
+                fasta=fasta,
+            )
+        except ValidationError as error:
+            LOG.error(error)
+            raise ValueError
+
+        samples_full_list: list = []
+        instrument_full_list: list = []
+        fasta_full_list: list = []
+        # Complete sample lists to the same length as fastq_r1:
+        for _ in range(len(fastq_r1)):
+            samples_full_list.append(case_id)
+            instrument_full_list.append(instrument_platform)
+            fasta_full_list.append(fasta)
```

Comment:
And we don't need a `for` loop, you can achieve the same using list comprehension:
```suggestion
samples_full_list: List[str] = [case_id] * len(fastq_r1)
instrument_full_list: List[str] = [instrument_platform] * len(fastq_r1)
fasta_full_list: List[str] = [fasta] * len(fastq_r1)
```

Reply:
Great!